src/p_map.cpp

```diff
@@ -4496,7 +4497,25 @@ AActor *P_LineAttack(AActor *t1, DAngle angle, double distance,
 	if (nointeract || (puffDefaults && puffDefaults->flags6 & MF6_NOTRIGGER)) tflags = TRACE_NoSky;
 	else tflags = TRACE_NoSky | TRACE_Impact;
 
-	if (!Trace(t1->PosAtZ(shootz), t1->Sector, direction, distance, MF_SHOOTABLE, 
+	// [MC] Check the flags and set the position according to what is desired.
+	// LAF_ABSPOSITION: Treat the offset parameters as direct coordinates.
+	// LAF_ABSOFFSET: Ignore the angle.
+
+	DVector3 tempos;
+	
+	if (flags & LAF_ABSPOSITION)
+		tempos = DVector3(offsetforward, offsetside, sz);
+	else if (flags & LAF_ABSOFFSET)
+		tempos = t1->Vec3Offset(offsetforward, offsetside, shootz);
+	else
+	{
+		double s = angle.Sin();
+		double c = angle.Cos();
+		tempos = t1->Vec3Offset(offsetforward * c + offsetside * s, offsetforward * s - offsetside * c, shootz);
```

Comment:
Usage of `Vec3Offset()` here and above is wrong. This function will add `shootz` to player's Z coordinate. Although `PosAtZ()` replaces the value with its argument.
You can verify this yourself by firing regular Doom pistol with no mod(s).
These two lines needs to be adjusted to use `Vec2OffsetZ()` function.

Reply:
Done.